File: TP1/TP1-exercice3/app/src/main/java/com/example/tp1_exercice3/SecondActivity.java
package com.example.tp1_exercice3;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

public class SecondActivity extends AppCompatActivity {
    private TextView quest1, quest2;
    private Button quitter;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_second);

        quest1 = findViewById(R.id.quest1);
        quest2 = findViewById(R.id.quest2);
        quitter = findViewById(R.id.quitter);

        quest1.setText(this.getIntent().getStringExtra("quest1"));
        quest2.setText(this.getIntent().getStringExtra("quest2"));

        quitter.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                finish();
            }
        });
    }
}

File: Projet_Controle_NO_ROOM/app/src/main/java/ma/ensa/projet/SecondActivity.kt
package ma.ensa.projet

import android.os.Build
import android.view.View
import android.view.animation.AnticipateOvershootInterpolator
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.animation.BounceInterpolator

class SecondActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_second)

        // Masquer la barre de statut
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {
            window.setDecorFitsSystemWindows(false)
        } else {
            window.decorView.systemUiVisibility = View.SYSTEM_UI_FLAG_FULLSCREEN
        }

        val title = findViewById<TextView>(R.id.welcomeTitle)
        val message = findViewById<TextView>(R.id.welcomeMessage)

        // Configuration initiale
        title.apply {
            scaleX = 0f
            scaleY = 0f
            alpha = 0f
            translationY = -200f
            rotation = -15f
        }

        message.apply {
            scaleX = 0f
            scaleY = 0f
            alpha = 0f
            translationY = 200f
            rotation = 15f
        }

        // Animation du titre
        title.animate()
            .alpha(1f)
            .translationY(0f)
            .rotation(0f)
            .scaleX(1.2f)
            .scaleY(1.2f)
            .setDuration(1200)
            .setInterpolator(AnticipateOvershootInterpolator(1.5f))
            .withEndAction {
                title.animate()
                    .scaleX(1f)
                    .scaleY(1f)
                    .setDuration(400)
                    .setInterpolator(BounceInterpolator())
                    .start()
            }
            .start()

        message.animate()
            .alpha(1f)
            .translationY(0f)
            .rotation(0f)
            .scaleX(1.2f)
            .scaleY(1.2f)
            .setStartDelay(600)
            .setDuration(1200)
            .setInterpolator(AnticipateOvershootInterpolator(1.5f))
            .withEndAction {
                message.animate()
                    .scaleX(1f)
                    .scaleY(1f)
                    .setDuration(400)
                    .setInterpolator(BounceInterpolator())
                    .start()
            }
            .start()
    }
}